db/schema.go

```diff
@@ -149,11 +148,19 @@ func (db *db) patchSchema(ctx context.Context, txn datastore.Txn, patchString st
 		if err != nil {
 			return err
 		}
+	}
+
+	newCollections, err := db.getAllCollections(ctx, txn)
+	if err != nil {
+		return err
+	}
 
-		newCollections = append(newCollections, col.Definition())
+	definitions := make([]client.CollectionDefinition, len(newCollections))
+	for i, col := range newCollections {
+		definitions[i] = col.Definition()
```

Comment:
tought: It would be nice to have a method `db.getAllCollectionDefinitions` that would return the definitions directly instead of having to loop through collections to get what we actually want.

Reply:
ah, I think that would only be used here no? And if it wasn't it may end up coupling totally unrelated parts of the codebase together - is not like this loop is particularly complex and in need of standardisation.

Leaving as is unless significant push back received.